### PR TITLE
Misc artifact balance changes - bonus fixes - bonus traits

### PR DESCRIPTION
**Pull request by DrDuckedGoose**
## About The Pull Request

Various changes to reel artifacts back in from being beyond ridiculous to just plainly ridiculous.
Also includes bonus fixes & content.

**See changelog for more details.**

## Why It's Good For The Game

There's a couple oddball traits that really impact the narrative / gameplay of a round. It's not uncommon for this effect to be 'too much' and negatively interact with the round.

New traits / content provide more active interest in the job. Additionally helps pad out more powerful traits.

## Testing Photographs and Procedure

<summary>Screenshots&Videos</summary>
<details>

``New UI``
image
image

``Anti-Clone``
image

``Delaminating``
image
image

``Dissapaiting``
This is 100%+ effectiveness.
image
This is 50% effectivness
image

``Marker``
image

``Absorbing``
image

``Honked``
image

``Emotional``
image

``Desynced``
image
image
image

``Hallucinogenic``
image
image
image

</details>

## Changelog
:cl:
tweak: Tweaks ``Hypodermic`` injections rates from ``5, 10, 15`` > (``7, 14 , 21`` * ``metabolization_rate``).
tweak: Changes ``C.D.E`` knockout from ``5 Seconds`` > ``2 Seconds``
tweak: ``Barreled`` will no longer always target east when not given a specific direction, instead it will pick a random direction.
tweak: ``Lamp`` now has a visible message which aids in identifying it in well-lit rooms
tweak: Changes logging ``xyz`` > ``area``
tweak: Change research point reward from ``2.3 * final price`` > ``2.5 * final price``
tweak: Change discovery point reward from ``350*final_price/initial_price`` > ``350*(final price/initial price)) * max(1, final price/1000)``
tweak: Aggressively lowers stun times on all related traits
tweak: Artifacts will now target the user too. The only way to avoid this is to be inside science & wearing safety-pinchers.
tweak: Slightly lower sentient buffs to arfiacts
tweak: Blacklist aura from destabilizing. 
tweak: Artifacts malfunction more often.
tweak: All artifacts get an extra slot for malfunctions, ``1`` > ``2``
tweak: Extend sentient play offer
tweak: Wall trait now points randomly east or north if it is 3x1
tweak: Increase bananium malfunction chance
tweak: Ghosts can take control of player-less artifacts
tweak: Artifact consoles now have a stability meter that stops spam-buying.
tweak: Sentient artifacts are more common, slightly,
tweak: Ghosts can activate haunted artifacts.
tweak: Tweaks labeler UI to show elements in alphabetical order.
tweak: Tweak console UI to have less clutter.
fix: Swaps over to AREACOORD for logging.
fix: Fix cheeky runtime on ``Weighted``. No longer checks all gloves for ``safety`` variable.
fix: Fix issue with ``weighted`` signal order preventing activation.
fix: Fix ``lamp`` trait runtiming
fix: Fix ``anti_artifact`` runtiming
fix: Fix fragile artifacts not breaking due to byond being beyond illogical. 
fix: Fix strip malfunction from stripping non-removable clothing..
add: Adds ``anti-clone`` malfunction. Creates an evil twin of the target.
add: Add better functions for adding malfunctions that support typing
add: Add exploding malfunction ``delaminating``, unique to bananium.
add: Add all artifact smoke screen trait.
add: Add Plasma & Bluespace trait, ``Marker`` trait, colors the target
add: Add all artifact malfunction ``absorbing``. Absorbs nearby gasses, suffocating small rooms. Will spit gasses back out when destroyed.
add: Add ``Honked`` activator. Only clowns, and clown related items, can activate this artifact.
add: Add ``Emotional`` major. Makes target perform a random emote, from a curated list.
add: Add ``Desynced`` minor. Makes the artifact disappear for a short period after activating.
add: Add ``Hallucinogenic`` malfunction. Makes the target experience a random hallucination.
add: Add bar to UI that shows thread stability.
/:cl:

Thanks to everyone who helped, especially @EvilDragonfiend

### CODE REVIEW

**Comment by EvilDragonfiend** [comment]
> tweak: Tweaks Hypodermic injections rates from 5, 10, 15 > 1, 3, 5

I believe this should be
`metabolisation rate * 3~7` then it means you get the amount of these chemicals as 3~7 ticks 1.2u to 2.8u for 0.4u metabolisation rate.
1u is way too short for most chemicals

**Comment by Tyranicranger4** [comment]
Does get_area(src) include both coordinates and the area similar to AREACOORD? Or is it just the area?

**Comment by EvilDragonfiend** [comment]
previously, 15u/10u/5u was equal to 38 ticks/25 ticks/13 ticks by 0.4u metabolisation rate base.
For now, 15 ticks(=6u), 10 ticks(=4u), 5 ticks(2u) will be very sane in general. I think you don't have to change the value.

**Comment by CydiaLamiales** [comment]
I think both the money payout and the research/discovery point rewards are too small. I even found myself loosing money after correctly identifying all the traits of an artefact which IMO doesn't make sense.

**Comment by itsmeow** [comment]
image

**Comment by DrDuckedGoose** [comment]
> image

Fixed in this PR 👍

**Comment by itsmeow** [comment]
Can you add a way for sentient artifacts to offer the slot to someone else?

**Comment by WhereAmO** [comment]
There's a neat thing with sentient slimes and ghosting. If you ghost from them, anyone, even you, can click the slime to become it again, this might be nice for sentient artifacts considering that sometimes they get stuck in strange places, and the current artifact goes "Oh, woe is me, I'll never be found again" and ghosts, but then gets found again later.

**Comment by itsmeow** [comment]
Tried xenoarch so I'm putting my feedback here

Comrade is spelt wrong
image
Anomaly is spelt wrong
image

Confused what this entire menu is for, it just has a ton of "requests" but it's not clear what they are at all
image

If you click the currently selected tab it does this
image

Why is this it's own tab? Just put the button by the credits display
image

This looks weird, why is there no comma/a space?
image

I don't like this button layout. Remove Purchase and the period, and then just move the shopping cart to the left side
image

This should be sorted alphabetically
image

**Comment by DrDuckedGoose** [comment]
made most of the sane changes, kept in future-content stuff, since science rework will probably kill it anyway.

File: tgui/packages/tgui/interfaces/XenoartifactConsole.js [hunk 1]
```diff
@@ -20,13 +21,22 @@ export const XenoartifactConsole = (props, context) => {
...
+            Thread stability 
+          </ProgressBar>
           <Section title={`Research and Development`} fluid
             buttons={(
               <Box fontFamily="verdana" inline bold>
                 <AnimatedNumber
                   value={points}
                   format={value => formatMoney(value)} />
-                {' credits'}
+                {',credits'}
```
Comment (itsmeow) [inline tgui/packages/tgui/interfaces/XenoartifactConsole.js]: pls no

File: tgui/packages/tgui/interfaces/XenoartifactConsole.js [hunk 2]
```diff
@@ -92,7 +102,7 @@ export const XenoartifactListingBuy = (props, context) => {
           {`${dialogue}`}
         </BlockQuote>
         <Button onClick={() => act(id)}>
-          {`Purchase: ${price} credits.`} <Icon name="shopping-cart" />
+          <Icon name="shopping-cart" />{`: ${price} credits`}
```
Comment (itsmeow) [inline tgui/packages/tgui/interfaces/XenoartifactConsole.js]: remove the colon

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui/interfaces/XenoartifactConsole.js [hunk 3]
```diff
@@ -92,7 +102,7 @@ export const XenoartifactListingBuy = (props, context) => {
           {`${dialogue}`}
         </BlockQuote>
         <Button onClick={() => act(id)}>
-          {`Purchase: ${price} credits.`} <Icon name="shopping-cart" />
```
Comment (itsmeow) [inline tgui/packages/tgui/interfaces/XenoartifactConsole.js:95]: Sorry I didn't mention this earlier actually but remove the space as well and move the icon into the button attribute for icons like most buttons are
icon="shopping-cart" will work the same

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui/interfaces/XenoartifactConsole.js (hunk 3 above)
Comment (DrDuckedGoose) [reply tgui/packages/tgui/interfaces/XenoartifactConsole.js:95]: Yep, added it to documentation images & CL

Comment (DrDuckedGoose) [reply tgui/packages/tgui/interfaces/XenoartifactConsole.js:95]: Also applied this to export menu.

**Comment by DrDuckedGoose** [comment]
Forgot to fix 'commrade` typo

**Review by itsmeow — APPROVED** [review]

**Comment by itsmeow** [comment]
Update wiki please

**Comment by DrDuckedGoose** [comment]
shit, changelog has a lie. Should be fine :trollface:

**Pull request merged**